src/Semantic/Api/Symbols.hs

```diff
@@ -49,39 +52,48 @@ legacyParseSymbols blobs = Legacy.ParseTreeSymbolResponse <$> distributeFoldMap
         tagToSymbol Tag{..}
           = Legacy.Symbol
           { symbolName = name
-          , symbolKind = kind
-          , symbolLine = fromMaybe mempty line
+          , symbolKind = pack (show kind)
```

Comment:
Rather than making the conversion here (and depending implicitly on the `Show` instance), can you make a conversion function in `Api.Bridge`, where all the conversion to wire formats happens? e.g. `instance APIConvert Kind ByteString`

Reply:
I am _really_ 👎 on touching that in this PR—`Legacy.Symbol` takes `Text` here, just like it’s always done, and so this is adhering to the interface we’re using in the wild. Changing it to a `ByteString` means we’ve got to think about, and deal with, encoding problems, and ultimately this has _always_ been fed by strings representing the names of constructors, it’s just that it was happening elsewhere before.

Reply:
If we want to revisit that stuff, I’m planning on a follow-up PR which will try to abstract the tagging approaches s.t. we do à la carte tagging via `semantic-tags` as well, and that might be a better time to look at it. But I’m not sure it’s a good idea to do so given that it’s working and depended upon.

Reply:
Sorry, I meant `APIConvert Kind Text`, not `ByteString`.

Reply:
All the above criticisms are totally valid - this shouldn’t be a ByteString conversion! I just want to consolidate the logic and make it explicit, rather than having these conversions occur implicitly and in two files.

Reply:
I was wrong- this conversion involves Kind, Text, and String, so APIConvert is the wrong place.

Reply:
We discussed this further offline, and as it’s not roundtripping, we’re gonna pass on this.